lib/logstash/outputs/elasticsearch.rb

```diff
@@ -178,6 +178,8 @@ class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
   #
   # - index: indexes a document (an event from Logstash).
   # - delete: deletes a document by id
+  # - create_unless_exists: creates a document, raises exception if no id is provided
```

Comment:
We should explicitly document that this is not supported by the HTTP client unless it does. I'm torn on that difference too, but it makes sense for the HTTP client to be a bit truer to the REST spec.